### DISCUSSION TITLE
Speed up woodpecker pipelines

### DISCUSSION THREAD

**Discussion by anbraten** [woodpecker-ci/woodpecker#2172] [General]
The pipelines executed for PRs in this repo could be improved to save resources and speed up:
- use more path filters
- only push images if requested for a PR (could be done by adding a label to the PR)
- only run build for amd64 architecture as cross-compiling is pretty slow
- ...

**Comment by qwerty287** [comment]
What do you think about skipping `test` and `web` pipelines on master pushes/merging PRs? They only contain linters and test and since almost everything is done via PRs the code should be checked by them already when a commit is pushed to the PR.

**Comment by anbraten** [comment]
Yeah sounds fine to me

**Comment by 6543** [comment]
did some bits at #1966

**Comment by qwerty287** [comment]
We did most for this:
- path filters for most steps
- docker pushes only with label
- only deploying, image pushing and codecov on pushes
- reduced archs for PRs

**Reply by anbraten** [reply]
last thing I can image would be #2295
